vowpalwabbit/core/include/vw/core/reductions/eigen_memory_tree.h

```diff
@@ -49,6 +49,24 @@ emt_feats emt_scale_add(float, const emt_feats&, float, const emt_feats&);
 
 emt_feats emt_router_eigen(std::vector<emt_feats>&, VW::rand_state&);
 
+template<typename I, typename R>
+void emt_shuffle(I first, I last, R* rng)
```

Comment:
Based on the implementation this will only work with a `VW::rand_state`, so I think we should just remove the template for `R` and use that type.

Reply:
Agreed, that's what I had originally but the compiler was yelling at me.

> invalid use of incomplete type ‘class VW::rand_state’

Any suggestions on this?